server/matcher/match_test.go

```diff
@@ -0,0 +1,134 @@
+package matcher
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/decred/dcrdex/server/account"
+	"github.com/decred/dcrdex/server/asset"
+	"github.com/decred/dcrdex/server/market/order"
+)
+
+var acct0 = account.AccountID{
+	0x22, 0x4c, 0xba, 0xaa, 0xfa, 0x80, 0xbf, 0x3b, 0xd1, 0xff, 0x73, 0x15,
+	0x90, 0xbc, 0xbd, 0xda, 0x5a, 0x76, 0xf9, 0x1e, 0x60, 0xa1, 0x56, 0x99,
+	0x46, 0x34, 0xe9, 0x1c, 0xec, 0x25, 0xd5, 0x40,
+}
+
+var (
+	marketOrders = []*order.MarketOrder{
+		{
+			Prefix: order.Prefix{
+				AccountID:  acct0,
+				BaseAsset:  asset.AssetDCR,
+				QuoteAsset: asset.AssetBTC,
+				OrderType:  order.MarketOrderType,
+				ClientTime: time.Unix(1566497653, 0),
+				ServerTime: time.Unix(1566497656, 0),
+			},
+			UTXOs:    []asset.UTXO{},
+			Side:     order.BuySide,
+			Quantity: 132413241324,
+			Address:  "DcqXswjTPnUcd4FRCkX4vRJxmVtfgGVa5ui",
+		},
+		{
+			Prefix: order.Prefix{
+				AccountID:  acct0,
+				BaseAsset:  asset.AssetDCR,
+				QuoteAsset: asset.AssetBTC,
+				OrderType:  order.MarketOrderType,
+				ClientTime: time.Unix(1566497654, 0),
+				ServerTime: time.Unix(1566497656, 0),
+			},
+			UTXOs:    []asset.UTXO{},
+			Side:     order.SellSide,
+			Quantity: 32413241324,
+			Address:  "DcqXswjTPnUcd4FRCkX4vRJxmVtfgGVa5ui",
+		},
+	}
+
+	limitOrders = []*order.LimitOrder{
+		{
+			MarketOrder: order.MarketOrder{
+				Prefix: order.Prefix{
+					AccountID:  acct0,
+					BaseAsset:  asset.AssetDCR,
+					QuoteAsset: asset.AssetBTC,
+					OrderType:  order.LimitOrderType,
+					ClientTime: time.Unix(1566497653, 0),
+					ServerTime: time.Unix(1566497656, 0),
+				},
+				UTXOs:    []asset.UTXO{},
+				Side:     order.BuySide,
+				Quantity: 102413241324,
+				Address:  "DcqXswjTPnUcd4FRCkX4vRJxmVtfgGVa5ui",
+			},
+			Rate:  0.13241324,
+			Force: order.StandingTiF,
+		},
+	}
+)
+
+func Test_shuffleQueue(t *testing.T) {
+	tests := []struct {
+		name string
+		in   []order.Order
+		out  []order.Order
+	}{
+		{
+			"1",
+			[]order.Order{
+				marketOrders[0],
+				marketOrders[1],
+				limitOrders[0],
+			},
+			[]order.Order{
+				marketOrders[0],
+				limitOrders[0],
+				marketOrders[1],
+			},
+		},
+	}
+	for _, tt := range tests {
+		// for _, o := range tt.in {
+		// 	t.Logf("%x", o.ID())
+		// }
+		t.Run(tt.name, func(t *testing.T) {
+			shuffleQueue(tt.in)
+			if !reflect.DeepEqual(tt.in, tt.out) {
```

Comment:
Just for fun, can we run at least one off these shuffles a couple times in a row on the same epoch and make sure nothing changes.

Reply:
Good call, will do.  There are a few more tests I was planning to add, but that wasn't one of them.